modules/hal_nordic/nrfx/Kconfig

```diff
@@ -34,31 +34,37 @@ config NRFX_EGU
 
 config NRFX_EGU0
 	bool "EGU0 instance"
+	default y if "$(dt_nodelabel_enabled,egu0)"
```

Comment:
Select relevant `NRFX_EGUx` symbols in `INTC_NRF_EGU` option instead, similarly to how e.g. the PWM driver does it: https://github.com/zephyrproject-rtos/zephyr/blob/99964fab45f7763bb623a020776609415512d3cd/drivers/pwm/Kconfig.nrfx#L9-L12